gensim/utils.py

```diff
@@ -376,11 +376,26 @@ def _save_specials(self, fname, separately, sep_limit, ignore, pickle_protocol,
                 elif isinstance(val, sparse_matrices) and val.nnz >= sep_limit:
                     separately.append(attrib)
 
+        def delete_attribute(object, attribute):
+            if hasattr(object, attribute):
+                asides[attribute] = getattr(object, attribute)
+                delattr(object, attribute)
+                
         # whatever's in `separately` or `ignore` at this point won't get pickled
         for attrib in separately + list(ignore):
-            if hasattr(self, attrib):
-                asides[attrib] = getattr(self, attrib)
-                delattr(self, attrib)
+            # As a result of maintaining backwards compatibility through __getattr__ after refactoring
+            # "syn0", "syn0norm", "vocab", and "index2word" out of Word2Vec to KeyedVectors,
+            # hasattr(self, "syn0") will return True but delattr(self, "syn0") will fail because it's
+            # not a true attribute of Word2Vec, Word2Vec.__getattr__ just reroutes it to KeyedVectors' attributes
+            if attrib in ["syn0", "syn0norm", "vocab", "index2word"]:
+                if type(self) == "Word2Vec":  # if saving a Word2Vec model, syn0, etc. are stored in self.kv
```

Comment:
Or we could use property descriptors and add setters and deleters for syn0norm, vocab etc, and perform those operations on `self.kv`. Wouldn't require any special condition checks anywhere, and we could add warnings that direct access to syn0/syn0norm would be removed in a future release.